@navikt/core/react/src/modal/Modal.tsx

```diff
@@ -79,111 +94,153 @@ type ModalLifecycle = {
+        }
       }
-      return rootElement !== undefined
-        ? () => rootElement as HTMLElement
-        : undefined;
-    };
+    }, [modalRef, open]);
+
+    useEffect(() => {
+      if (!modalRef.current) return;
+      if (modalRef.current.open) document.body.classList.add(bodyClass); // In case `open` is true initially
+
+      const observer = new MutationObserver(() => {
+        if (modalRef.current?.open) document.body.classList.add(bodyClass);
+        else document.body.classList.remove(bodyClass);
+      });
+      observer.observe(modalRef.current, {
+        attributes: true,
+        attributeFilter: ["open"],
+      });
+      return () => {
+        observer.disconnect();
+        document.body.classList.remove(bodyClass); // In case modal is unmounted before it's closed
+      };
+    }, [modalRef]);
 
     return (
-      <ReactModal
-        {...rest}
-        parentSelector={getParentSelector()}
-        style={style}
-        isOpen={open}
+      <dialog
         ref={mergedRef}
-        className={cl("navds-modal", className)}
-        overlayClassName={cl("navds-modal__overlay", overlayClassName)}
-        shouldCloseOnOverlayClick={shouldCloseOnOverlayClick}
-        shouldCloseOnEsc={shouldCloseOnEsc}
-        onRequestClose={(e) => onModalCloseRequest(e)}
-        aria={{
-          describedby: ariaDescribedBy,
-          labelledby: ariaLabelledBy,
-          modal: ariaModal,
+        className={cl("navds-modal", className, {
+          "navds-modal--polyfilled": needPolyfill,
+          "navds-modal--autowidth": !width,
+        })}
+        style={{ ...style, width }}
+        onCancel={(event) => {
+          // FYI: onCancel fires when you press Esc
+          if (onBeforeClose && onBeforeClose() === false) {
+            event.preventDefault();
+          } else if (onCancel) onCancel(event);
         }}
-        contentLabel={contentLabel}
+        aria-labelledby={
+          !ariaLabelledby && !rest["aria-label"] && header
+            ? ariaLabelId
+            : ariaLabelledby
+        }
+        {...rest}
       >
-        {children}
-        {closeButton && (
-          <Button
-            className={cl("navds-modal__button", {
-              "navds-modal__button--shake": shouldCloseOnOverlayClick,
-            })}
-            size="small"
-            variant="tertiary-neutral"
-            ref={buttonRef}
-            onClick={onClose}
-            icon={<XMarkIcon title="Lukk modalvindu" />}
-          />
-        )}
-      </ReactModal>
+        <ModalContext.Provider
+          value={{
+            closeHandler: getCloseHandler(modalRef, header, onBeforeClose),
+          }}
+        >
+          {header && (
+            <ModalHeader>
+              {header.label && (
+                <Detail className="navds-modal__label">{header.label}</Detail>
+              )}
+              <Heading
+                size={header.size ?? "medium"}
+                level="2"
```

Comment:
Siden Modal er sin egen context, bør denne være `h1` (`level="1"`)?

Reply:
Usikker på hva du mener med "sin egen context" i denne konteksten (🥁), men så vidt jeg vet og har lest, så skal det helst bare være én h1 på en side.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#avoid_using_multiple_h1_elements_on_one_page

Når det er sagt, så er det kanskje litt tvilsomt å hardkode til h2 uansett, så kanskje det skulle vært en prop?

Hva mener @it-vegard?

Reply:
Har tolket `dialog` slik at den oppfører seg som en "ny" side skjermlesere. Siden skjermleser ikke har tilgang til innholdet bak `dialog`, vil det da være mulig å starte på `h1`.